In [1]:
from main import DA,df2,results,accuracy

In [2]:
DA

Variable     Type  NA_values
0                         ID   object          0
1                Customer_ID   object          0
2                      Month   object          0
3                       Name   object       9985
4                        Age   object          0
5                        SSN   object          0
6                 Occupation   object          0
7              Annual_Income   object          0
8      Monthly_Inhand_Salary  float64      15002
9          Num_Bank_Accounts    int64          0
10           Num_Credit_Card    int64          0
11             Interest_Rate    int64          0
12               Num_of_Loan   object          0
13              Type_of_Loan   object      11408
14       Delay_from_due_date    int64          0
15    Num_of_Delayed_Payment   object       7002
16      Changed_Credit_Limit   object          0
17      Num_Credit_Inquiries  float64       1965
18                Credit_Mix   object          0
19          Outstanding_Debt   object          0
20  Credit_Utilization_Ratio  float64          0
21        Credit_History_Age   object       9030
22     Payment_of_Min_Amount   object          0
23       Total_EMI_per_month  float64          0
24   Amount_invested_monthly   object       4479
25         Payment_Behaviour   object          0
26           Monthly_Balance   object       1200
27              Credit_Score   object          0

In [3]:
df2.head(10)

Name  Annual_Income  Monthly_Inhand_Salary  Interest_Rate  \
0    AaronMaashoh      1911412.0            1824.843333            3.0   
1    AaronMaashoh      1911412.0            1824.843333            3.0   
2    AaronMaashoh      1911412.0            1824.843333            3.0   
3    AaronMaashoh      1911412.0            1824.843333            3.0   
4    AaronMaashoh      1911412.0            1824.843333            3.0   
5    AaronMaashoh      1911412.0            1824.843333            3.0   
6    AaronMaashoh      1911412.0            1824.843333            3.0   
7    AaronMaashoh      1911412.0            1824.843333            3.0   
8  RickRothackerj      3484784.0            3037.986667            6.0   
9  RickRothackerj      3484784.0            3037.986667            6.0   

   Delay_from_due_date  Num_of_Delayed_Payment  Num_Credit_Inquiries  \
0                  3.0                     7.0                   4.0   
1                 -1.0                     0.0                   4.0   
2                  3.0                     7.0                   4.0   
3                  5.0                     4.0                   4.0   
4                  6.0                     0.0                   4.0   
5                  8.0                     4.0                   4.0   
6                  3.0                     8.0                   4.0   
7                  3.0                     6.0                   4.0   
8                  3.0                     4.0                   2.0   
9                  7.0                     1.0                   2.0   

  Credit_Mix  Outstanding_Debt  Credit_Utilization_Ratio  Credit_History_Age  \
0                      80998.0                 26.822620               265.0   
1       Good           80998.0                 31.944960               265.0   
2       Good           80998.0                 28.609352               267.0   
3       Good           80998.0                 31.377862               268.0   
4       Good           80998.0                 24.797347               269.0   
5       Good           80998.0                 27.262259               270.0   
6       Good           80998.0                 22.537593               271.0   
7       Good           80998.0                 23.933795               271.0   
8       Good           60503.0                 24.464031               319.0   
9       Good           60503.0                 38.550848               320.0   

   Total_EMI_per_month  Amount_invested_monthly  Monthly_Balance Credit_Score  
0            49.574949             8.041530e+15     3.124941e+16         Good  
1            49.574949             1.182802e+16     2.846292e+16         Good  
2            49.574949             8.169952e+13     3.312099e+15         Good  
3            49.574949             1.994581e+15     2.234513e+16         Good  
4            49.574949             4.142015e+16     3.414892e+16         Good  
5            49.574949             6.243017e+16     3.404792e+15         Good  
6            49.574949             1.783441e+15     2.445653e+15         Good  
7            49.574949             2.478522e+16     3.581242e+16     Standard  
8            18.816215             1.042918e+14     4.706906e+16     Standard  
9            18.816215             4.039124e+15     4.845912e+15         Good

In [4]:
results.head(10)

Name  FICO_Punctuation  Altman_Z-score  Altman_Punctuation  \
0    AaronMaashoh               680    2.601203e+13                1000   
1    AaronMaashoh               785    2.705625e+13                1000   
2    AaronMaashoh               805    2.181422e+12                1000   
3    AaronMaashoh               705    1.579507e+13                1000   
4    AaronMaashoh               785    5.279874e+13                1000   
5    AaronMaashoh               755    4.880874e+13                1000   
6    AaronMaashoh               775    2.897753e+12                1000   
7    AaronMaashoh               805    4.143910e+13                1000   
8  RickRothackerj               820    1.852553e+13                1000   
9  RickRothackerj               750    3.752243e+12                1000   

   Final_Punctuation Final_Score Original_Score  
0              840.0        Good           Good  
1              892.5        Good           Good  
2              902.5        Good           Good  
3              852.5        Good           Good  
4              892.5        Good           Good  
5              877.5        Good           Good  
6              887.5        Good           Good  
7              902.5        Good       Standard  
8              910.0        Good       Standard  
9              875.0        Good           Good

In [5]:
accuracy

0.50204

FICO Model
pay_his = ['Delay_from_due_date','Num_of_Delayed_Payment'] # Payment History 35%
debt = ['Outstanding_Debt'] # Amount Debt 30%
cred_his = ['Credit_History_Age'] # Credit History 10%
cred_req = ['Num_Credit_Inquiries'] # New Credit Requests 15%
typ_cred = ['Credit_Mix'] # Type of Credits 10%

Plus 
rate = ['Interest_Rate'] # Interest Rate 5%
cred_ut = ['Credit_Utilization_Ratio'] # Credit Utilization 5%